docs/content/custom-replication-system.md

```diff
@@ -115,24 +118,24 @@ public class CubeColorSendSystem : CustomSpatialOSSendSystem<SpatialOSCubeColor>
         for (var i = 0; i < cubeColorData.Length; i++)
         {
             var entityId = cubeColorData.SpatialEntityIds[i];
-            var eventSender = cubeColorData.CubeColorEventSenders[i];
-            var changeColorEvents = eventSender.GetChangeColorEvents();
+            var component = cubeColorData.CubeColors[i];
+            var changeColorEvents = cubeColorData.ChangeColorEventSenders[i].Events;
 
             if(changeColorEvents.Count == 0)
             {
                 continue;
             }
 
-            var update = new global::Playground.CubeColor.Update();
-
+            var update = new SchemaComponentUpdate(component.ComponentId);
+            var eventsObj = update.GetEvents();
             foreach(var event in changeColorEvents)
             {
-                update.changeColor.Add(global::Generated.Playground.ColorData.ToSpatial(event));
+                var eventObj = eventsObj.AddObject(1); // NOTE: 1 corresponds to the event index in schema. 
+                global::Generated.Playground.ColorData.Serialization.Serialize(event, eventObj)
             }
 
-            SpatialOSCubeColorTranslation.SendComponentUpdate(worker.Connection, entityId, update);
-
-            eventSender.ClearChangeColorEvents();
+            worker.Connection.SendComponentUpdate(entityId, new ComponentUpdate(update));
+            changeColorEvents.Clear();
```

Comment:
in the `TriggerColorChangeSystem` you don't  need to do all that. You simply add the event to the events list 🤔 
which one should be the proposed way?

Reply:
This isn't an example of sending events normally - this is for if you wanted to write custom serialisation logic for events 😉

Reply:
ahh, got it 👍